doc/source/reference/random/generator.rst

```diff
@@ -36,11 +36,109 @@ Simple random data
 
 Permutations
 ============
+
+The methods `Generator.shuffle`, `Generator.permutation` and
+`Generator.permuted` are used for randomly permuting an input array.
+The following describes the important differences among these methods.
+
+In-place vs. copy
+~~~~~~~~~~~~~~~~~
+The main difference between `Generator.shuffle` and `Generator.permutation`
+is that `Generator.shuffle` operates in-place, while `Generator.permutation`
+returns a copy.
+
+By default, `Generator.permuted` returns a copy.  To operate in-place with
+`Generator.permuted`, pass the same array as the first argument *and* as
+the value of the ``out`` parameter.  For example,
+
+    >>> x
+    array([[ 0,  1,  2,  3,  4],
+           [ 5,  6,  7,  8,  9],
+           [10, 11, 12, 13, 14]])
+    >>> y = rg.permuted(x, axis=1, out=x)
+    >>> x
+    array([[ 1,  0,  2,  4,  3],  # random
+           [ 6,  7,  8,  9,  5],
+           [10, 14, 11, 13, 12]])
```

Comment:
These examples fail doctests: `x` and `rg` are not defined. Personally I think I perfer to lead with the table rather than put it at the end, but it is hard to see without rendered docs.

Reply:
Thanks @mattip, I like the suggestion to move the table to the top.  I'll make that change in my next update.

Reply:
> That still allows the compiler to optimize based on knowing that itemsize is sizeof(npy_intp)?

From Sebastian and from some further reading: yes, gcc can, in fact, replace a `memcpy` call with inline code.

Reply:
I updated the docs.